src/cpp/hu.bme.mit.inf.modes3.components.traindetector/traindetector/traindetector.cpp

```diff
@@ -0,0 +1,23 @@
+#include "traindetector.h"
+
+static int ID_COUNTER = 0;
+
+const TrainDetection &TrainDetector::handleEdge(SensorSide side, EdgeDirection dir, time_t timestamp) {
+    for (TrainDetection &td : this->detections) {
+        if (td.step(side, dir, timestamp)) {
```

Comment:
Extract the bool return to a new variable first, then use the variable in the condition. The code will be more readable then.